pjsip/include/pjsua2/media.hpp

```diff
@@ -426,6 +426,75 @@ typedef std::vector<AudioMedia*> AudioMediaVector;
 /** Array of Audio Media */
 typedef std::vector<AudioMedia> AudioMediaVector2;
 
+/**
+ * This structure describes a media frame.
+ */
+struct MediaFrame
+{
+    pjmedia_frame_type   type;      /**< Frame type.                        */
+    string               buf;       /**< Frame buffer content.              */
+    unsigned             size;      /**< Frame size in bytes.               */
+
+public:
+    /**
+     * Default constructor
+     */
+    MediaFrame()
+    : type(PJMEDIA_FRAME_TYPE_NONE),
+      size(0)
+    {}
+};
+
+/**
+ * Audio Media Port.
+ */
+class AudioMediaPort : public AudioMedia
```

Comment:
How about adding a method for unregistering the media port?

Reply:
The parent class already has that method (it's protected though, but since it's mostly for internal purpose it should be fine).